retrieval/target.go

```diff
@@ -359,9 +364,15 @@ func (t *Target) scrape(sampleAppender storage.SampleAppender) (err error) {
 	}()
 
 	for samples := range t.ingestedSamples {
-		for _, s := range samples {
-			s.Metric.MergeFromLabelSet(baseLabels, clientmodel.ExporterLabelPrefix)
-			sampleAppender.Append(s)
+		if t.proxy {
+			for _, s := range samples {
+				sampleAppender.Append(s)
```

Comment:
That should be added my the scraping side, as a target doesn't know what a scraper wants label wise - it could be run by a different team for example.

I plan to add the relabel logic to various places in the system, and metrics ingestion time is one of them which should cover this. I'd suggest not holding up this review on that.